src/foam/core/FObject.js

```diff
@@ -808,6 +808,12 @@ foam.CLASS({
 
     function compareTo(other) {
       if ( other === this ) return 0;
+
+      // Use comparator of other if it is not an FObject.
+      if ( ! foam.core.FObject.isInstance(other) ) {
```

Comment:
My understanding is that `FObject` is the last thing we would put on the left. Hence, if `other` is an `Object`, then `foam.util.compare(this, other)` will eventually lead to `-foam.Object.compare(other, this)`.